### PR TITLE
azure-instance-using-default-auth-tf-and-ansible-support TF&Ansible support

### PR DESCRIPTION
**Pull request by Tohar-orca**
**Proposed Changes**
- add detection rules for azure-instance-using-default-auth-tf-and-ansible-support under Ansible and Terraform

I submit this contribution under the Apache-2.0 license.

### CODE REVIEW

File: assets/queries/ansible/azure/azure_instance_using_basic_authentication/metadata.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "id": "e2d834b7-8b25-4935-af53-4a60668dcbe0",
+  "queryName": "Azure Instance Using Basic Authentication",
+  "severity": "HIGH",
+  "category": "Best Practices",
+  "descriptionText": "Azure Instances should use SSH Key instead of basic authentication",
+  "descriptionUrl": "https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/linux_virtual_machine#admin_ssh_key",
```
Comment (cxMiguelSilva) [inline assets/queries/ansible/azure/azure_instance_using_basic_authentication/metadata.json:7]:
```suggestion
  "descriptionUrl": "https://docs.ansible.com/ansible/latest/collections/azure/azcollection/azure_rm_virtualmachine_module.html#parameter-linux_config/disable_password_authentication",
```
Update the descriptioURL to the alsible docs

File: assets/queries/ansible/azure/azure_instance_using_basic_authentication/query.rego [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package Cx
+
+import future.keywords.if
+
+CxPolicy[result] {
+	vm := input.document[i].playbooks[k].azure_rm_virtualmachine
+    is_linux_vm(vm)
+    not vm.ssh_public_keys
```
Comment (cxMiguelSilva) [inline assets/queries/ansible/azure/azure_instance_using_basic_authentication/query.rego:8]:
```suggestion
    not vm.linux_config.disable_password_authentication
```
Update the query to check the disable_password_authentication

File: assets/queries/terraform/azure/azure_instance_using_basic_authentication/query.rego [hunk 3]
```diff
@@ -0,0 +1,35 @@
+package Cx
+
+import future.keywords.if
+import data.generic.terraform as tf_lib
+
+CxPolicy[result] {
+	vm := input.document[i].resource.azurerm_virtual_machine[name]
+    object.get(vm, "os_profile_linux_config", false)
+	not vm.admin_ssh_key
```
Comment (cxMiguelSilva) [inline assets/queries/terraform/azure/azure_instance_using_basic_authentication/query.rego:9]:
```suggestion
	not vm.disable_password_authentication
```

File: assets/queries/terraform/azure/azure_instance_using_basic_authentication/query.rego [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+		"searchKey": sprintf("%s[%s].admin_ssh_key", [resource_type, name]),
+		"issueType": "MissingAttribute",
+		"keyExpectedValue": sprintf("'%s[%s]' should be using SSH keys for authentication", [resource_type, name]),
+		"keyActualValue": sprintf("'%s[%s]' is using username and password for authentication", [resource_type, name]),
+	}
+}
+
+CxPolicy[result] {
+	vm := input.document[i].resource.azurerm_linux_virtual_machine[name]
+	not vm.admin_ssh_key
```
Comment (cxMiguelSilva) [inline assets/queries/terraform/azure/azure_instance_using_basic_authentication/query.rego:24]:
```suggestion
    not vm.disable_password_authentication
```

**Review by cxMiguelSilva — CHANGES_REQUESTED** [review]
Hi @Tohar-orca,
In the first place, sorry for the late response.
Regarding the proposed changes, after talking with our application security team, the queries created for Ansible and Terraform should follow the approach of the AzureReseourceManagement security query to reduce false positives. So the security queries should check the disable_password_authentication parameter. 
Also, update the keyExpectedValues/keyActualValues to go accordingly to the query.

File: assets/queries/ansible/azure/azure_instance_using_basic_authentication/query.rego [hunk 5]
```diff
@@ -0,0 +1,24 @@
+package Cx
+
+import future.keywords.if
+
+CxPolicy[result] {
+	vm := input.document[i].playbooks[k].azure_rm_virtualmachine
+    is_linux_vm(vm)
+    not vm.ssh_password_enabled == false
```
Comment (cxMiguelSilva) [inline assets/queries/ansible/azure/azure_instance_using_basic_authentication/query.rego:8]:
```suggestion
    not vm.linux_config.disable_password_authentication == false
```

**Review by cxMiguelSilva — CHANGES_REQUESTED** [review]

File: assets/queries/ansible/azure/azure_instance_using_basic_authentication/query.rego (hunk 5 above)
Comment (Tohar-orca) [reply assets/queries/ansible/azure/azure_instance_using_basic_authentication/query.rego:8]: @cxMiguelSilva I can't seem to find `linux_config` and `disable_password_authentication` in the Ansible documentation, are you positive this is correct configuration?
https://docs.ansible.com/ansible/2.7/modules/azure_rm_virtualmachine_module.html

Comment (cxMiguelSilva) [reply assets/queries/ansible/azure/azure_instance_using_basic_authentication/query.rego:8]: I have found it here
https://docs.ansible.com/ansible/latest/collections/azure/azcollection/azure_rm_virtualmachine_module.html#parameter-linux_config/disable_password_authentication
Let me know what you think

Comment (Tohar-orca) [reply assets/queries/ansible/azure/azure_instance_using_basic_authentication/query.rego:8]: I see that's a newer version of Ansible.. I'll edit the rule to include both cases then

Comment (cxMiguelSilva) [reply assets/queries/ansible/azure/azure_instance_using_basic_authentication/query.rego:8]: Ok thank you so much!

**Review by cxMiguelSilva — APPROVED** [review]

**Pull request merged**